### ISSUE TITLE
Automatic digitization of a closed line, polygon, or filled object

### ISSUE THREAD

**Issue by foret37** [ankitrohatgi/WebPlotDigitizer#199]
Your automatic digitization feature is very nice. Would it be possible to include the ability to automatically digitize an object. At present, the program will find points for an object (as defined by a fill or a closed line) but the points are not organized as a path around the object.
Thanks for your consideration.
Best,
Mark

**Comment by nbehrnd** [comment]
Mark's idea ignites an other one: maybe not «only» use WPD to estimate diameter and perimeter of objects on a map, but to calculate / approximate its surface as long as one referenced an external bar.  One example in topographic maps is Google Maps Area Calculator ([one example](https://3planeta.com/googlemaps/google-maps-area-calculator-tools.html).  But it equally could be more than this, e.g. to determine the typical grain sizes in a microscopic sample like in polarization microscopy, too.  As a speculation: similar like WPD already guesses where a blue line transitions into the background, maybe WPD equally may determine (an initial) grain / cell boundery, too.

Equally with best regard,
Norwid

**Comment by ankitrohatgi** [comment]
Yes, this is something I want to get to eventually. At the moment, you can manually add a polygon to calculate area, perimeter etc. (under "measurements") but no automatic methods yet. 🔜

**Comment by foret37** [comment]
Thanks for the reply.
Best,
Mark
